discord/guild.go

```diff
@@ -196,47 +221,16 @@ func (guild *GuildState) handleTrackedMembers(sm *SessionManager, delay int, han
 			}
 		}
 	}
-	wg := sync.WaitGroup{}
-	waitForHigherPriority := false
-
-	if delay > 0 {
-		log.Printf("Sleeping for %d seconds before applying changes to users\n", delay)
-		time.Sleep(time.Second * time.Duration(delay))
-	}
-
-	for priorityQueue.Len() > 0 {
-		p := heap.Pop(priorityQueue).(PrioritizedPatchParams)
-
-		if p.priority > 0 {
-			waitForHigherPriority = true
-			log.Printf("User %s has higher priority: %d\n", p.patchParams.Userdata.GetID(), p.priority)
-		} else if waitForHigherPriority {
-			//wait for all the other users to get muted/unmuted completely, first
-			//log.Println("Waiting for high priority user changes first")
-			wg.Wait()
-			waitForHigherPriority = false
-		}
 
-		wg.Add(1)
+	// Close channel because all users were sent
+	close(userChan)
```

Comment:
Is this `close()` guaranteed to happen after all events have been pulled from the channel? My channel closing knowledge is rusty, just wanted to make sure `close` is only relevant after all events are guaranteed to have been pulled by workers.

Reply:
The channel only "truly closes" when its also empty. I've created an a plaground that helps illustrate (an test) this!
https://play.golang.org/p/Qgbdn6BlcBL

Reply:
The only purpose on closing the channel is to top the workers.
If you comment the `close` line in the playground, the worker will not close (and never run the defer)